d2lib/d2.go

```diff
@@ -68,14 +68,28 @@ func compile(ctx context.Context, g *d2graph.Graph, opts *CompileOptions) (*d2ta
 			return nil, err
 		}
 
-		constantNears := d2near.WithoutConstantNears(ctx, g)
+		constantNearGraphs := d2near.WithoutConstantNears(ctx, g)
+
+		// run core layout for constantNears
+		for _, tempGraph := range constantNearGraphs {
+			if tempGraph == nil {
+				continue
+			}
+
+			nearObject := tempGraph.Root.ChildrenArray[0]
+			nearObject.Parent = tempGraph.Root
```

Comment:
is this doing anything?

Reply:
yep, seems like `dagre` need this to do the layout work

Reply:
it seems like this belongs in `WithoutConstantNears`.

Reply:
fixed